### ISSUE TITLE
Convert CITATION.md to CITATION.cff

### ISSUE THREAD

**Issue by benjaminrose** [sncosmo/sncosmo#302]
GitHub added a new feature to promote software citation, https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-citation-files. I think we should update our file format and go with the flow.

**Comment by kboone** [comment]
Looks like a good idea to me. There is a zenodo release for sncosmo to cite, although it is very out of date. We should probably update the zenodo release before we add this.

@kbarbary Were you just doing the zenodo releases manually? There is a github integration now, but it doesn't look like that is currently set up for sncosmo on zenodo.

**Comment by benjaminrose** [comment]
In #231 we decided to go with the all-version zenodo DOI. We should be be able to make a CITATION.cff with that now. We should also update zenodo; my preference is to add an automatic system.

**Comment by benjaminrose** [comment]
Using https://github.com/citation-file-format/doi2cff, I was able to make a .cff file (see below). It pulled the v1.4.0 DOI (10.581/zenodo.168220) rather than the all-version DOI (10.5281/zenodo.592747). I think we should change the DOI in this to the all-versions DOI. Anything else that needs updating?

```
# YAML 1.2
# Metadata for citation of this software according to the CFF format (https://citation-file-format.github.io/)
cff-version: 1.0.3
message: If you use this software, please cite it using these metadata.
# FIXME title as repository name might not be the best name, please make human readable
title: 'sncosmo/sncosmo: v1.4.0'
doi: 10.5281/zenodo.168220
# FIXME splitting of full names is error prone, please check if given/family name are correct
authors:
- given-names: Kyle
  family-names: Barbary
  affiliation: UC Berkeley
- given-names: rbiswas4
  family-names: ''
- given-names: Danny
  family-names: Goldstein
  affiliation: UC Berkeley / LBNL C3
- given-names: Steve
  family-names: Rodney
  affiliation: University of South Carolina
- given-names: Saurabh
  family-names: Jha
  name-particle: W
- given-names: Michael
  family-names: Wood-Vasey
- given-names: Caroline
  family-names: Sofiatti
  affiliation: LBNL
- given-names: Ulrich
  family-names: Feindt
  affiliation: Oskar Klein Centre
- given-names: Brian
  family-names: Friesen
  affiliation: Lawrence Berkeley National Laboratory
- given-names: Tom
  family-names: Barclay
- given-names: R.
  family-names: Thomas
  name-particle: C.
  affiliation: Berkeley Lab (NERSC)
- given-names: Matt
  family-names: Craig
  affiliation: Minnesota State University Moorhead
- given-names: Geert
  family-names: Barentsen
  affiliation: NASA
version: 1.4.0
date-released: 2016-11-23
repository-code: https://github.com/sncosmo/sncosmo
license: other-open
```

**Comment by kbarbary** [comment]
I think I had an integration set up at some point in the distant past. It might have needed to be reconfigured at some point, or I might have removed it.

This was all before there was an "all-version" zenodo DOI, and there was only the per-version DOIs. I think the value of individual versions as a citation target is low, so I didn't make much effort to maintain them.

**Comment by benjaminrose** [comment]
While digging around more, I am unsure if the `version` and `date-released` keys are required. It looks they [are not](https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md), but that they [use to be](https://github.com/citation-file-format/citation-file-format/blob/1.1.0/README.md#cff-version-required).

**Comment by benjaminrose** [comment]
I think we can use this file, once we make sure the author list is correct: 

```
# YAML 1.2
# Metadata for citation of this software according to the CFF format (https://citation-file-format.github.io/)
cff-version: 1.2.0
message: "If you use SNCosmo, please cite it using these metadata."
title: SNCosmo
doi: 10.5281/zenodo.592747
# FIXME splitting of full names is error prone, please check if given/family name are correct
authors:
- given-names: Kyle
  family-names: Barbary
  affiliation: UC Berkeley
- given-names: rbiswas4
  family-names: ''
- given-names: Danny
  family-names: Goldstein
  affiliation: UC Berkeley / LBNL C3
- given-names: Steve
  family-names: Rodney
  affiliation: University of South Carolina
- given-names: Saurabh
  family-names: Jha
  name-particle: W
- given-names: Michael
  family-names: Wood-Vasey
- given-names: Caroline
  family-names: Sofiatti
  affiliation: LBNL
- given-names: Ulrich
  family-names: Feindt
  affiliation: Oskar Klein Centre
- given-names: Brian
  family-names: Friesen
  affiliation: Lawrence Berkeley National Laboratory
- given-names: Tom
  family-names: Barclay
- given-names: R.
  family-names: Thomas
  name-particle: C.
  affiliation: Berkeley Lab (NERSC)
- given-names: Matt
  family-names: Craig
  affiliation: Minnesota State University Moorhead
- given-names: Geert
  family-names: Barentsen
  affiliation: NASA
- given-names: Rose
  family-names: Benjamin
  affiliation: Duke University
  orcid: 0000-0002-1873-8973
repository-code: https://github.com/sncosmo/sncosmo
license: other-open
```

Also, if we change to `.cff` we should think about adding a validation to our GitHub actions, https://github.com/marketplace/actions/cff-validator.

**Comment by kboone** [comment]
We should add all of the contributors to this. We have been maintaining a list of contributors on the docs/about page here: https://github.com/sncosmo/sncosmo/blob/master/docs/about.rst

It would be good to double check this against the list of contributors here too: https://github.com/sncosmo/sncosmo/graphs/contributors

**Comment by benjaminrose** [comment]
I am good with combining all three lists with @kbarbary first, then alphabetical.

**Comment by kboone** [comment]
Sounds good. Do you want to go ahead and do it or should I?

**Comment by kboone** [comment]
I'll go ahead and start working on this. I want to get the 2.6 release out ASAP because of #314 which is blocking some other projects for me.

**Comment by kboone** [comment]
Ok I made a PR with a first go at this. I put @kbarbary first, and then everybody else alphabetical. I tried to fill in affiliations/orcids wherever possible. I'm not really sure what to do for people who have left the field.

**Comment by benjaminrose** [comment]
We should also add a `.cff` validator action, https://github.com/marketplace/actions/cff-validator. Do we want to change all of this at once, or do these admin tasks another PR?

**Comment by kboone** [comment]
I don't think that we should use that action at the moment. It has only been out for 8 days and has not received much attention on GitHub. I'm worried about maintaining it, and it also requires a full R environment to run that I'm not familiar with. Let's hold off on adding an automatic validator for now. We can deal with that in a separate PR in a few months when the CFF ecosystem is more developed.

**Comment by kboone** [comment]
I think that I fixed the Zenodo integration, so the next release should automatically get published over there.

**Comment by kboone** [comment]
Merged #315 that implements this.
